sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+package com.microsoft.appcenter.utils;
+
+import android.app.Activity;
+import android.os.Bundle;
+import android.os.Handler;
+
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+
+import java.util.Timer;
+import java.util.TimerTask;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@PrepareForTest({Handler.class})
+public class ApplicationLifecycleListenerTest {
+
+    private Handler mHandlerMock;
+    private ApplicationLifecycleListener mApplicationLifecycleListenerMock;
+    private Activity mActivityMock;
+
+    @Before
+    public void setUp() {
+        mHandlerMock = mock(Handler.class);
+        mApplicationLifecycleListenerMock = new ApplicationLifecycleListener(mHandlerMock);
+        mActivityMock = mock(Activity.class);
+        doAnswer(new Answer<Void>() {
+
+            @Override
+            public Void answer(InvocationOnMock invocation) {
+                ((Runnable) invocation.getArguments()[0]).run();
+                return null;
+            }
+        }).when(mHandlerMock).postDelayed(any(Runnable.class), anyLong());
+        doAnswer(new Answer<Void>() {
+
+            @Override
+            public Void answer(InvocationOnMock invocation) {
+                ((Runnable) invocation.getArguments()[0]).run();
+                return null;
+            }
+        }).when(mHandlerMock).removeCallbacks(any(Runnable.class));
+    }
+
+    @After
+    public void tearDown() {
+        mHandlerMock = null;
+        mApplicationLifecycleListenerMock = null;
+        mActivityMock = null;
+    }
```

Comment:
Seems useless

Reply:
Doesn't it have `stopMocking` or something like this?